app/src/main/java/org/schabi/newpipe/fragments/BaseStateFragment.java

```diff
@@ -223,15 +228,30 @@ protected boolean onError(final Throwable exception) {
         if (exception instanceof ReCaptchaException) {
             onReCaptchaException((ReCaptchaException) exception);
             return true;
-        } else if (exception instanceof ContentNotAvailableException) {
-            showError(getString(R.string.content_not_available), false);
-            return true;
         } else if (ExceptionUtils.isNetworkRelated(exception)) {
             showError(getString(R.string.network_error), true);
             return true;
+        } else if (exception instanceof GeographicRestrictionException) {
+            showError(getString(R.string.georestricted_content), false);
+            return true;
+        } else if (exception instanceof SoundCloudGoPlusException) {
+            showError(getString(R.string.soundcloud_go_plus_content), false);
+            return true;
+        } else if (exception instanceof PrivateContentException) {
+            showError(getString(R.string.private_content), false);
+            return true;
+        } else if (exception instanceof YoutubeMusicPremiumContentException) {
+            showError(getString(R.string.youtube_music_premium_content), false);
+            return true;
+        } else if (exception instanceof PaidContentException) {
+            showError(getString(R.string.paid_content), false);
+            return true;
         } else if (exception instanceof ContentNotSupportedException) {
             showError(getString(R.string.content_not_supported), false);
             return true;
+        } else if (exception instanceof ContentNotAvailableException) {
+            showError(getString(R.string.content_not_available), false);
+            return true;
```

Comment:
Please move the `ContentNotAvailableException` branch before the `ContentNotSupportedException`branch to keep the logical order (`ContentNotSupportedException` is not a `ContentNotAvailableException`)